tests/unit/components/boolean-input-test.js

```diff
@@ -44,6 +44,30 @@ describeComponent(
       ])
     })
 
+    it('shows error messages when store.showErrorMessages and showErrorMessages are true', function () {
+      component.set('showErrorMessages', false)
+      component.set('store.showAllErrors', true)
+      expect(component.get('showErrorMessages')).to.be.truthy
+    })
```

Comment:
Also, it appears you're working with `showErrorMessages` (note the plural) when the actual property I think you want is `showErrorMessage`